tooling/lsp/src/lib.rs

```diff
@@ -209,23 +214,38 @@ fn byte_span_to_range<'a, F: files::Files<'a> + ?Sized>(
 }
 
 pub(crate) fn resolve_workspace_for_source_path(file_path: &Path) -> Result<Workspace, LspError> {
-    let package_root = find_file_manifest(file_path);
-
-    let toml_path = package_root.ok_or_else(|| {
-        LspError::WorkspaceResolutionError(format!(
-            "Nargo.toml not found for file: {:?}",
-            file_path
-        ))
-    })?;
-
-    let workspace = resolve_workspace_from_toml(
-        &toml_path,
-        PackageSelection::All,
-        Some(NOIR_ARTIFACT_VERSION_STRING.to_string()),
-    )
-    .map_err(|err| LspError::WorkspaceResolutionError(err.to_string()))?;
-
-    Ok(workspace)
+    if let Some(toml_path) = find_file_manifest(file_path) {
+        resolve_workspace_from_toml(
+            &toml_path,
+            PackageSelection::All,
+            Some(NOIR_ARTIFACT_VERSION_STRING.to_string()),
+        )
+        .map_err(|err| LspError::WorkspaceResolutionError(err.to_string()))
+    } else {
+        let Some(parent_folder) = file_path.parent().and_then(|f| f.file_name()).and_then(|file_name_os_str| file_name_os_str.to_str()) else {
+            return Err(LspError::WorkspaceResolutionError(format!(
+                "Could not resolve parent folder for file: {:?}",
+                file_path
+            )))
+        };
+        let assumed_package = Package {
+            version: None,
+            compiler_required_version: Some(NOIR_ARTIFACT_VERSION_STRING.to_string()),
+            root_dir: PathBuf::from(parent_folder),
+            package_type: PackageType::Binary,
+            entry_path: PathBuf::from(file_path),
+            name: CrateName::from_str(parent_folder)
+                .map_err(|err| LspError::WorkspaceResolutionError(err.to_string()))?,
+            dependencies: BTreeMap::new(),
+        };
+        let workspace = Workspace {
+            root_dir: PathBuf::from(parent_folder),
+            members: vec![assumed_package],
+            selected_package_index: Some(0),
+            is_assumed: false,
```

Comment:
It looks like the new `is_assumed` field is always `false`. Was it meant to be `true` here?

Reply:
Good catch, thank you...
b4430b0: chore: correct is_assumed value